packages/remix/src/build.ts

```diff
@@ -174,9 +174,81 @@ export const build: BuildV2 = async ({
     pkg.dependencies?.['@remix-run/dev'] ||
     pkg.devDependencies?.['@remix-run/dev'];
 
+  const serverBundlesMap = new Map<string, ConfigRoute[]>();
+  const resolvedConfigsMap = new Map<ConfigRoute, ResolvedRouteConfig>();
+
+  // Read the `export const config` (if any) for each route
+  const project = new Project();
+  const staticConfigsMap = new Map<ConfigRoute, BaseFunctionConfig | null>();
+  for (const route of remixRoutes) {
+    const routePath = join(remixConfig.appDirectory, route.file);
+    let staticConfig = getConfig(project, routePath);
+    if (staticConfig && isHydrogen2) {
+      console.log(
+        'WARN: `export const config` is currently not supported for Hydrogen v2 apps'
+      );
+      staticConfig = null;
+    }
+    staticConfigsMap.set(route, staticConfig);
+  }
+
+  for (const route of remixRoutes) {
+    const config = getResolvedRouteConfig(
+      route,
+      remixConfig.routes,
+      staticConfigsMap,
+      isHydrogen2
+    );
+    resolvedConfigsMap.set(route, config);
+  }
+
+  // Figure out which routes belong to which server bundles
+  // based on having common static config properties
+  for (const route of remixRoutes) {
+    if (isLayoutRoute(route.id, remixRoutes)) continue;
+
+    const config = resolvedConfigsMap.get(route);
+    if (!config) {
+      throw new Error(`Expected resolved config for "${route.id}"`);
+    }
+    const hash = calculateRouteConfigHash(config);
+
+    let routesForHash = serverBundlesMap.get(hash);
+    if (!Array.isArray(routesForHash)) {
+      routesForHash = [];
+      serverBundlesMap.set(hash, routesForHash);
+    }
+
+    routesForHash.push(route);
+  }
+
+  const serverBundles: ServerBundle[] = Array.from(
+    serverBundlesMap.entries()
+  ).map(([hash, routes]) => {
+    const runtime = resolvedConfigsMap.get(routes[0])?.runtime ?? 'nodejs';
+    return {
+      serverBuildPath: isHydrogen2
+        ? relative(entrypointFsDirname, remixConfig.serverBuildPath)
+        : `${relative(
+            entrypointFsDirname,
+            dirname(remixConfig.serverBuildPath)
+          )}/build-${runtime}-${hash}.js`,
+      routes: routes.map(r => r.id),
+    };
+  });
+
+  // If the project is not relying on split configuration,
+  // then don't inject the `@vercel/remix-run-dev` fork
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏This looks like a comment for line 251, not here